src/v/redpanda/admin/api-doc/transform.json

```diff
@@ -205,6 +236,23 @@
           "description": "The offset within the input topic that has been committed"
         }
       }
-    }
+    },
+      "transform_metadata_patch": {
+          "id": "transform_metadata_patch",
+          "description": "A partial update to transform metadata",
```

Comment:
Is there a way to document the semantics? I feel like patch requests are sometimes hard to understand. For example would be good to say things like:

* environment map is overridden (so no deep patches)
* If not present then it is unchanged